ipc/api/src/subnet.rs

```diff
@@ -67,18 +67,50 @@ pub enum SupplyKind {
 
 #[derive(Clone, Debug, Serialize, Deserialize, PartialEq, Eq)]
 pub struct ConstructParams {
+    // optional fields, set to default if not provided
+    pub ipc_gateway_addr: Option<Address>,
+    pub power_scale: Option<i8>,
+    pub consensus: Option<ConsensusType>,
+    pub majority_percentage: Option<u8>,
+    pub active_validators_limit: Option<u16>,
```

Comment:
Can you explain why these needed to be made optional because of the `--dry-run`?

Reply:
no no, it's just previously they are hardcoded or scattered around. From the old code, I inferred that they are optional, so I put them in one place.